### PR TITLE
Clean pro finder.mobile premise editing

### PR DESCRIPTION
**Pull request by firtap**

### CODE REVIEW

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using CommunityToolkit.Mvvm.Input;
+
+namespace CleanProFinder.Mobile.ViewModels;
+
+public partial class ServiceUserAddPremiseViewModel : ObservableObject
+{
+    private readonly IDialogService _dialogService;
+    private readonly IUserPremiseService _userPremisesService;
+
+    public ServiceUserAddPremiseViewModel(IDialogService dialogService, IUserPremiseService userPremisesServic)
```
Comment (Losdrew) [inline CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs:13]: Fix premise service parameter name (in other files too)

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        if (float.TryParse(Square, out float result))
+        {
+            squareAsFloat = result;
+        }
+        else
+        {
+            await _dialogService.ShowAlertAsync("Incorect Square Format", String.Empty, "Ok");
+            return;
+        }
```
Comment (Losdrew) [inline CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs:41]: Use numeric keyboard instead

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        ServiceResponse response = 
+            await _userPremisesService.AddServiceUserPremiseAsync(squareAsFloat, Description, Address);
+
+        if (response.IsSuccess)
+        {
+            await Shell.Current.GoToAsync("..");
+            return;
+        }
+        await _dialogService.ShowErrorAlertAsync("Adding Premise failed", response.Error);
```
Comment (Losdrew) [inline CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs:51]: Add new line

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+using CleanProFinder.Mobile.Services;
+using CleanProFinder.Shared.ServiceResponseHandling;
+using CommunityToolkit.Mvvm.Input;
+
+namespace CleanProFinder.Mobile.ViewModels;
+
+public partial class ServiceUserAddPremiseViewModel : ObservableObject
+{
+    private readonly IDialogService _dialogService;
+    private readonly IUserPremiseService _userPremisesService;
```
Comment (Losdrew) [inline CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs:11]: Rename to _userPremiseService (in other files too)

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    [RelayCommand]
+    private async void StartEditing()
+    {
+        IsEditing = true;
+    }
+
+    private async void LoadPremise(string premiseId)
+    {
+        ServiceResponse<OwnPremiseFullInfoDto> response =
+           await _userPremisesService.GetServiceUserPremiseFullInfoAsync(new Dictionary<string, object> { { "premiseId", premiseId } });
```
Comment (Losdrew) [inline CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs:55]: Move dictionary initialization to a separate line

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs (hunk 3 above)
Comment (Losdrew) [reply CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs:51]: All words in error titles should start from capital letter (in other files too)

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+        if (float.TryParse(Square, out float result))
+        {
+            squareAsFloat = result;
+        }
+        else
+        {
+            await _dialogService.ShowAlertAsync("Incorect Square Format", String.Empty, "Ok");
+            return;
+        }
```
Comment (Losdrew) [inline CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs:81]: Use numeric keyboard instead

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+        await _dialogService.ShowErrorAlertAsync("Updating Premise failed", response.Error);
+    }
+
+    [RelayCommand]
+    private async void DeletePremise()
+    {
+        ServiceResponse response =
+           await _userPremisesService.DeleteServiceUserPremiseAsync(new Dictionary<string, object> { { "premiseId", PremiseId } });
```
Comment (Losdrew) [inline CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs:100]: Move dictionary initialization to a separate line

**Review by Losdrew — CHANGES_REQUESTED** [review]

File: CleanProFinder/CleanProFinder.Mobile/CleanProFinder.Mobile.csproj [hunk 8]
```diff
@@ -74,11 +74,17 @@
...
+	  <MauiXaml Update="Views\RolePage.xaml">
+	    <Generator>MSBuild:Compile</Generator>
+	  </MauiXaml>
+	  <MauiXaml Update="Views\ServiceUserAddPremisePage.xaml">
+	    <Generator>MSBuild:Compile</Generator>
+	  </MauiXaml>
+	  <MauiXaml Update="Views\ServiceUserEditPremisePage.xaml">
+	    <Generator>MSBuild:Compile</Generator>
+	  </MauiXaml>
+	</ItemGroup>
```
[the remark this answers is not in the document]
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/CleanProFinder.Mobile.csproj:87]: Fixed

File: CleanProFinder/CleanProFinder.Mobile/Services/IUserPremise.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿using CleanProFinder.Shared.Dto.Premises;
+using CleanProFinder.Shared.ServiceResponseHandling;
+
+namespace CleanProFinder.Mobile.Services;
+
+public interface IUserPremiseService
+{
+    Task<ServiceResponse> AddServiceUserPremiseAsync(float square, string description, string address);
+    Task<ServiceResponse> EditServiceUserPremiseAsync(Guid id, float square, string description, string address);
+    Task<ServiceResponse<IEnumerable<OwnPremiseShortInfoDto>>> GetServiceUserPremiseListAsync();
```
[the remark this answers is not in the document]
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/Services/IUserPremise.cs:10]: Fixed

File: CleanProFinder/CleanProFinder.Mobile/Services/IUserPremise.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+    Task<ServiceResponse<OwnPremiseFullInfoDto>> GetServiceUserPremiseFullInfoAsync(Dictionary<string, object> payload);
```
[the remark this answers is not in the document]
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/Services/IUserPremise.cs:11]: Fixed

File: CleanProFinder/CleanProFinder.Mobile/Services/UserPremiseService.cs [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+using CleanProFinder.Shared.ServiceResponseHandling;
+
+namespace CleanProFinder.Mobile.Services;
+
+public class UserPremiseService : IUserPremiseService
+{
+    private const string CreateServiceUserPremiseEndpoint = "api/premise/create";
+    private const string EditServiceUserPremiseEndpoint = "api/premise/edit";
+    private const string GetServiceUserPremiseListEndpoint = "api/premise/my-premises";
+    private const string GetServiceUserPremiseFullInfoEndpoint = "api/premise/full-info";
```
[the remark this answers is not in the document]
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/Services/UserPremiseService.cs:11]: Fixed

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs (hunk 1 above)
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs:13]: Fixed

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs (hunk 6 above)
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs:81]: Fixed

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs (hunk 7 above)
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs:100]: Fixed

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs (hunk 5 above)
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserEditPremiseViewModel.cs:55]: Fixed

File: CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs (hunk 3 above)
Comment (firtap) [reply CleanProFinder/CleanProFinder.Mobile/ViewModels/ServiceUserAddPremiseViewModel.cs:51]: Fixed

**Pull request merged**